### PR TITLE
Minor grammar improvements

### PR DESCRIPTION
**Pull request by subpop**
This PR contains a few minor grammatical changes to make the sentences flow a bit smoother. I tried to preserve the sentence structure as much as possible, while still improving the grammar.

### CODE REVIEW

**Comment by swilmet** [comment]
Thanks! I've modified the commits to remove some of your changes that I didn't really like.

And I've pushed a couple other commits to improve things differently. For example "the bases of GLib core" -> "the basics of GLib core" (not "the foundations").

And I've added you to the acknowledgment section.

File: src/oop.tex [hunk 1]
```diff
@@ -147,17 +147,17 @@ \subsubsection{GObject Introspection Annotations}
...
+Note that the \lstinline{g_return} macros should be used only for the entry points of a class, that is, in its public functions. You can usually assume that the parameters passed to a \lstinline{static} function are valid, especially the \emph{self} parameter. However, it is sometimes useful to check an argument of a \lstinline{static} function with \lstinline{g_assert()}, to make the code more robust and self-documented.
 
 \subsubsection{Coding Style}
-Finally, it is worth explaining a few things about the coding style. You are encouraged to use early on the same coding style for your project, because it can be something difficult to change afterwards. If every program has different code conventions, it's a nightmare for someone willing to contribute.
+Finally, it is worth explaining a few things about the coding style. You are encouraged early on to use the same coding style for your project, because it can be something difficult to change afterwards. If every program has different code conventions, it's a nightmare for someone willing to contribute.
```
Comment (swilmet) [inline src/oop.tex]: I didn't merge this change. "early on" should be "attached" to "to use the same coding style", not "encouraged". The important thing is to use early on the same coding style, not being early on encouraged to do so. If you see the difference. But maybe you have a better explanation for why you made that change.

File: src/oop.tex [hunk 2]
```diff
@@ -207,9 +207,9 @@ \section{GObject}
...
+GObject also simplifies the event-driven programming paradigm, with signals and properties. A property is basically an instance variable with a notify signal that is emitted when its value changes. Creating a GObject signal or property is a nice way to implement the Observer design pattern; that is, one or several objects \emph{observing} state changes of another object, by connecting function callbacks. The object \emph{emitting} the signal is not aware of which objects \emph{receive} the signal. GObject just keeps track of the list of callbacks to call. So adding a signal permits decoupling classes.
```
Comment (swilmet) [inline src/oop.tex]: I prefer "permits to decouple classes", not "permits decoupling classes". "to decouple" is more an action, i.e. taking one class design and refactoring it to another design with signals, to decouple the classes.

File: src/oop.tex [hunk 3]
```diff
@@ -207,9 +207,9 @@ \section{GObject}
...
 
 It is recommended to create GObject classes for writing a GLib/GTK+ application. Unfortunately the code is a little verbose, because the C language is not object-oriented. Boilerplate code is needed for some features, but don't be afraid, there are tools and scripts to generate the boilerplate.
 
 The GObject reference documentation contains introductory chapters: ``Concepts'' and ``Tutorial'', available at:\\
-\url{https://developer.gnome.org/gobject/stable/}
+\url{https://developer.gnome.org/gobject/stable/}.
```
Comment (swilmet) [inline src/oop.tex]: And I didn't merge this change either. If the document is printed, maybe someone will think that the dot is part of the URL.

File: src/further-reading.tex [hunk 4]
```diff
@@ -27,22 +27,22 @@ \section{GTK+ and GIO}
...
 
   \item For guidelines about OOP specifically, see \emph{Object-Oriented Design Heuristics}, by Arthur~Riel \cite{oop-book}.
 
   \item An excellent source of information is the website of Martin~Fowler: refactoring, agile methodology, code design, ...\\
   \url{http://martinfowler.com/}
 \end{itemize}
 
-More related to GNOME, Havoc~Pennington's articles have good advises worth the reading, including ``\emph{Working on Free Software}'', ``\emph{Free software UI}'' and ``\emph{Free Software Maintenance: Adding Features}'':\\
+More related to GNOME, Havoc~Pennington's articles have good advice worth reading, including ``\emph{Working on Free Software}'', ``\emph{Free software UI}'' and ``\emph{Free Software Maintenance: Adding Features}'':\\
```
Comment (swilmet) [inline src/further-reading.tex]: "good advice" -> ok, merged.
But I prefer "worth the reading". The grammar is correct, AFAIK. Maybe it's a difference of personal writing style.

Anyway, all the other changes are merged, thanks again! It indeed improves the text. I'll try to not reproduce some of the errors.

File: src/oop.tex (hunk 2 above)
Comment (399man) [inline src/oop.tex]: My English grammar says:
- permit _verb_+ing (ie. doing something)
- permit something/somebody to _verb_ (ie. someone to do something)

That is if the object of "permit" is a verb, then it is "permit _verb+ing_", but if the object is something or somebody, then it is "something/somebody to _verb_".

The same applies to at least to advise, allow, forbid, recommend, permit, require.

Comment (swilmet) [inline src/oop.tex]: "permits to decouple" returns more results in Google than "permits decoupling".

"It permits to decouple classes" sounds better to me.
"It permits decoupling classes" is more ambiguous. "decoupling classes" can be read as "decoupled classes", or "loose coupling between classes". Whereas "to decouple classes" implies an action, doing a refactoring, it's a more "active" sentence.

Comment (subpop) [inline src/oop.tex]: How about "permits the programmer to decouple classes from one another."?

**Pull request closed without merging**